networking/eth2/src/main/java/tech/pegasys/artemis/networking/eth2/Eth2NetworkBuilder.java

```diff
@@ -58,7 +60,8 @@ public Eth2Network build() {
     // Build core network and inject eth2 handlers
     final P2PNetwork<?> network = buildNetwork();
 
-    return new Eth2Network(network, eth2PeerManager, eventBus, chainStorageClient);
+    return DiscoveryNetwork.create(
```

Comment:
What about wrapping it the other way:  
```suggestion
    final P2PNetwork discoveryNetwork = DiscoveryNetwork.create(network, config);
    return new Eth2Network(discoveryNetwork, eth2PeerManager, eventBus, chainStorageClient);
```

Reply:
Yeah that makes sense.